core-plugins/src/main/java/io/cdap/plugin/batch/aggregator/function/ConcatDistinct.java

```diff
@@ -44,41 +47,25 @@ public ConcatDistinct(String fieldName, Schema fieldSchema) {
 
   @Override
   public void initialize() {
-    concatString = "";
+    values = new LinkedHashSet<>();
   }
 
   @Override
   public void mergeValue(StructuredRecord record) {
     if (record.get(fieldName) != null) {
       String value = record.get(fieldName);
-      if (value != null && !concatString.contains(value)) {
-        if (firstString) {
-          concatString = value;
-          firstString = false;
-          return;
-        }
-        concatString = String.format("%s, %s", concatString, value);
-      }
+      values.add(value);
```

Comment:
I think LinkedHashSet allows one null value. Do you want null value?

Reply:
We don't want nulls. We check for nulls in line 55 of this file before entering this code block.